bpy_speckle/properties/scene.py

```diff
@@ -11,6 +11,51 @@
     IntProperty,
 )
 
+class SelectionStateHack:
+    selected_user_id : str = None
+    selected_stream_id : str = None
+    selected_branch_id : str = None
+    selected_commit_id : str = None
+
+    @staticmethod
+    def get_item_id_by_index(collection, index):
+        # print(list(collection.items()))
+        selected_index = int(index)
+        for index, (key, item) in enumerate(collection.items()):
+            if index == selected_index:
+                return item.id
+        return None
+    
+    @staticmethod
+    def get_item_index_by_id(collection, id):
```

Comment:
Would it be possible for you to decorate these functions with the correct typing for the parameters and return type, please.

e.g.
```py
def restore_selection_state(speckle : SpeckleSceneSettings) -> None:
```

```py
def get_item_index_by_id(collection : SpeckleBranchObject, id: str) -> str:
```

etc.

This may prompt you to perform a few additional null checks (in which case, a simple `raise Exception("Useful message...")` etc should suffice)

I can do this step my self if you prefer, but it would help speed things up as I'm rather busy this week.

Ideally, every function would be typed, and code have zero typing warnings, but some compromises may be acceptable (particularly when we look at Blender's Properties that don't support correct typing)

So if you find a warning that doesn't have an obvious solution, leave it, we can supress the warning or I can fix.

---

If your using VS Code, I recommend using the [Python](https://marketplace.visualstudio.com/items?itemName=ms-python.python) extension. It will give you the type hinting and warnings.